src/plugins/usage_collection/server/usage_counters/usage_counters_service.ts

```diff
@@ -152,32 +156,46 @@ export class UsageCountersService {
     );
   }
 
-  private createUsageCounter = (type: string): UsageCounter => {
-    if (this.counterSets.get(type)) {
-      throw new Error(`Usage counter set "${type}" already exists.`);
+  private createUsageCounter = (
+    domainId: string,
+    soType: UsageCounterSavedObjectType = SERVER_COUNTERS_SAVED_OBJECT_TYPE
+  ): IUsageCounter => {
+    if (this.counterSets.get(domainId)) {
+      throw new Error(`Usage counter set "${domainId}" already exists.`);
     }
 
     const counterSet = new UsageCounter({
-      domainId: type,
+      domainId,
       counter$: this.source$,
+      soType,
     });
 
-    this.counterSets.set(type, counterSet);
+    this.counterSets.set(domainId, counterSet);
```

Comment:
`domainId` could clash for N different `soType`s.

Maybe this quick fix?
```suggestion
    this.counterSets.set(`${soType}:${domainId}`, counterSet);
```

But then we need to change `getUsageCounterByDomainId` to receive `soType` as well.

Reply:
I won't, as the service [enforces](https://github.com/elastic/kibana/pull/187064/files#r1657243338) to have unique `domainId` across all types.